MaterialComponent.js

```diff
@@ -53,23 +57,30 @@ export default class MaterialComponent extends Component {
     }
     return classText;
   }
+
   // Components must implement this method for their specific DOM structure
   materialDom(props) {
     return h("div", Object.assign({}, props), props.children);
   }
+
   render() {
     this.buildClassName();
     // Fetch a VNode
     const componentProps = this.props;
+    const userDefinedClasses =
+      componentProps.className || componentProps.class || "";
     if (componentProps.class) {
-      // We delete class prop here so that any sub node's class doesn't get over-ridden from this
+      // We delete class props and add them later in the final
+      // step so every component does not need to handle user specified classes.
+      delete componentProps.className;
       delete componentProps.class;
     }
     const element = this.materialDom(componentProps);
     element.attributes = element.attributes || {};
-    // Fix for className
-    element.attributes.class = this.getClassName(element);
-    // element.attributes.className = this.getClassName(element);
+
+    element.attributes.className = `${userDefinedClasses} ${this.getClassName(
```

Comment:
The `this.getClassName(element)` should be included in the logic I included in my review of the `userDefinedClasses` part. This will make sure that we don't have duplicate class names.

```JavaScript
const userDefinedClasses =
  ((componentProps.className || "") + " " + (componentProps.class || ""));
...
element.attributes.className =
  (userDefinedClasses + " " + this.getClassName(element))
  .split(' ') // split the class names into an array
  .filter((value, index, self) => self.indexOf(value) === index && value !== "") // Unique + exclude empty class names
  .join(' '); // Merge them into a string again;
```

Reply:
wondering if looking for unique classNames become too much for every render? WDYT?

Reply:
I don't think it would be a problem except if you have a ludicrous amount of class names and you call `render()` every second, but then I would think there are other problems that would occur before this becomes an issue (e.g. the DOM rendering has problems keeping up).

Breakdown of the methods used:
- `join` should be fairly inexpensive.
- `split` should also be inexpensive.
- `filter` is just a for loop that creates a new array with a condition. Highly doubt this would be an issue at all.
- `indexOf` should be fine too (https://jsperf.com/array-indexof-vs-set-has) but is probably the one that is the most expensive of them all.

Reply:
Done